x/rollapp/keeper/rollapp.go

```diff
@@ -245,3 +245,55 @@ func (k Keeper) IsRollappStarted(ctx sdk.Context, rollappId string) bool {
 	_, found := k.GetLatestStateInfoIndex(ctx, rollappId)
 	return found
 }
+
+func (k Keeper) FilterRollapps(ctx sdk.Context, f func(types.Rollapp) bool) []types.Rollapp {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.RollappKeyPrefix))
+	iterator := sdk.KVStorePrefixIterator(store, []byte{})
+	defer iterator.Close() // nolint: errcheck
+
+	var result []types.Rollapp
+	for ; iterator.Valid(); iterator.Next() {
+		var val types.Rollapp
+		k.cdc.MustUnmarshal(iterator.Value(), &val)
+		if f(val) {
+			result = append(result, val)
+		}
+	}
+	return result
+}
+
+func FilterNonVulnerable(b types.Rollapp) bool {
+	return !b.Frozen
+}
+
+func FilterVulnerable(b types.Rollapp) bool {
+	return b.Frozen
+}
+
+func (k Keeper) MarkRollappAsVulnerable(ctx sdk.Context, rollappId string) bool {
+	r, found := k.GetRollapp(ctx, rollappId)
+	if !found {
+		return false
+	}
+	r.Frozen = true
```

Comment:
when setting a rollapp to frozen, we have more logic to run (e.g as in `HandleFraud`)
probably should be wrapped and reused